### PR TITLE
nearest neighbor :queen no longer goes out of bounds

### PR DESCRIPTION
**Pull request by gottacatchenall**
**What the pull request does**   
fixes the bug in #53 

**Type of change**   
Please indicate the relevant option(s)

- [x] :bug: Bug fix (non-breaking change which fixes an issue)
- [ ] :sparkle: New feature (non-breaking change which adds functionality)
- [ ] :boom: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] :book: This change requires a documentation update

**Checklist**
- [ ] The changes are documented
  - [ ] The docstrings of the different functions describe the arguments, purpose, and behavior 
  - [ ] There are examples in the documentation website
- [ ] The changes are tested
- [ ] The changes **do not** modify the behavior of the previously existing functions
  - If they **do**, please explain why and how in the introduction paragraph
- [ ] For **new contributors** - my name and information are added to `.zenodo.json`
- [ ] The `Project.toml` field `version` has been updated
  - Change the *last* number for a `v0.0.x` series release, a bug fix, or a performance improvement
  - Change the *middle* number for additional features that *do not* break the current test suite (unless you fix a bug in the test suite)
  - Change the *first* number for changes that break the current test suite

### CODE REVIEW

File: src/classify.jl [hunk 1]
```diff
@@ -102,15 +102,24 @@ function label(mat, neighborhood = :rook)
     # run through the matrix and make clusters
     for j in axes(mat, 2), i in axes(mat, 1)
         if isfinite(mat[i, j])
-            same = [i - n[1] > 0 && j - n[2] > 0 && mat[i - n[1], j - n[2]] == mat[i, j] for n in neighbors]
```
Comment (mkborregaard) [inline src/classify.jl:105]: Wouldn't it be sifficient with something like
```
m,k = size(mat)
same = [0 < i - n[1] < m && 0 < j - n[2] > k && mat[i - n[1], j - n[2]] == mat[i, j] for n in neighbors]
```
```

Comment (mkborregaard) [reply src/classify.jl:105]: we could use the `m` and `k` instead of the axes too

Comment (tpoisot) [reply src/classify.jl:105]: @gottacatchenall I think it's a good suggestion - let's do this and I'll merge

Comment (gottacatchenall) [reply src/classify.jl:105]: at some point i tried replacing my fix with the above and it threw a different error, i'll look into it

Comment (tpoisot) [reply src/classify.jl:105]: If your fix makes the bug go away, let's go with your fix!

**Comment by tpoisot** [comment]
@gottacatchenall I've updated the release number, you can merge when all the tests pass - do you know how to tag the new release using the bot?

**Comment by gottacatchenall** [comment]
do i just make a new release? doesn't look like i can run the tagbot workflow @tpoisot

**Pull request merged**